### PR TITLE
fix last practiced info for library items

### LINKED ISSUE #7: Fix "last practiced" info for library items

### PR DESCRIPTION
**Pull request by matthiasemde**
Closes #7

### CODE REVIEW

File: app/src/main/java/app/musikus/database/daos/SectionDao.kt [hunk 1]
```diff
@@ -173,6 +169,26 @@ abstract class SectionDao(
...
+            "AND NOT EXISTS (" +
+            "SELECT 1 FROM section AS s " +
+            "WHERE s.library_item_id = section.library_item_id " +
+            "AND s.session_id IN (" +
+            "SELECT id FROM session " +
+            "WHERE deleted=0 " +
+            ") " +
+            "AND datetime(SUBSTR(s.start_timestamp, 1, INSTR(s.start_timestamp, '[') - 1)) > datetime(SUBSTR(section.start_timestamp, 1, INSTR(section.start_timestamp, '[') - 1)) " +
+            ")"
+    )
```
Comment (mipro98) [inline app/src/main/java/app/musikus/database/daos/SectionDao.kt:190]:
```suggestion
    @Query("SELECT * " +
            "FROM section " +
            "WHERE section.library_item_id IN (:itemIds) " +    // filter by itemIds
            "AND section.session_id IN (" +                     // filter non-deleted sessions
                "SELECT id FROM session " +
                "WHERE deleted=0 " +
            ")" +
            "AND NOT EXISTS (" +                                // filter out all sections that have a later section for the same item
                "SELECT 1 FROM section AS s " +
                "WHERE s.library_item_id = section.library_item_id " +
                "AND s.session_id IN (" +
                    "SELECT id FROM session " +
                    "WHERE deleted=0 " +
                ") " +
                "AND datetime(SUBSTR(s.start_timestamp, 1, INSTR(s.start_timestamp, '[') - 1)) > datetime(SUBSTR(section.start_timestamp, 1, INSTR(section.start_timestamp, '[') - 1)) " +
            ")"
    )
```

Can we add a bit more structure and explanaition to the query?

File: app/src/main/java/app/musikus/ui/library/LibraryScreen.kt [hunk 2]
```diff
@@ -543,7 +544,7 @@ fun LibraryUiItem(
                     },
                 )
                 Text(
-                    text = "last practiced: yesterday",
+                    text = "last practiced: " +  (lastPracticedDate?.musikusFormat(DateFormat.DAY_MONTH_YEAR) ?: "never"),
```
Comment (mipro98) [inline app/src/main/java/app/musikus/ui/library/LibraryScreen.kt:547]: Is this really the right place for human-readable conversion of `ZonedDateTime` object? What if we want to add more logic, e.g. "yesterday", "last sunday", "today" in the future? Wouln't be the ViewModel the right place to implement this *UI logic*?

**Review by mipro98 — CHANGES_REQUESTED** [review]
LGTM, only minor comments.

File: app/src/main/java/app/musikus/ui/library/LibraryScreen.kt (hunk 2 above)
Comment (matthiasemde) [reply app/src/main/java/app/musikus/ui/library/LibraryScreen.kt:547]: This will be solved with #4

**Review by mipro98 — APPROVED** [review]

**Pull request merged**